### PR TITLE
Implemented: functionality to all the rule pages (#255)

### PR DESCRIPTION
**Pull request by amansinghbais**
### Related Issues
 

 Related Issue #255

 ### Short Description and Why It's Useful
 
Added the functionality to the following pages
- Safety-stock page
- Store pickup page
- Shipping page

 ### Screenshots of Visual Changes before/after (If There Are Any)
 

 **IMPORTANT NOTICE** - Remember to add changelog entry

 ### Contribution and Currently Important Rules Acceptance
 

 - [x] I read and followed [contribution rules](https://github.com/hotwax/threshold-management#contribution-guideline)

### CODE REVIEW

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/UpdateProductFiltersModal.vue [hunk 1]
```diff
@@ -166,6 +166,8 @@ function isSelected(value: string) {
 async function saveFilters() {
   const rule = JSON.parse(JSON.stringify(props.rule))
 
+  if(!rule.ruleConditions) rule.ruleConditions = []
+
```
[the remark this answers is not in the document]
Comment (amansinghbais) [reply src/components/UpdateProductFiltersModal.vue:168]: Fixed this.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/RuleItem.vue [hunk 2]
```diff
@@ -206,8 +211,12 @@ async function editSafetyStock() {
     },
     {
       text: translate('Update'),
-      handler: async (data) => {
+      handler: async (data: any) => {
         if(data.safetyStock) {
+          if(data.safetyStock < 0) {
+            showToast(translate("Safety stock should be greater then 0."));
```
Comment (ymaheshwari1) [inline src/components/RuleItem.vue:217]:
```suggestion
            showToast(translate("Safety stock should be greater then or equal to 0."));
```

File: src/components/RuleItem.vue [hunk 3]
```diff
@@ -160,6 +160,11 @@ async function editThreshold() {
       text: translate('Update'),
       handler: async(data) => {
         if(data.threshold) {
+          if(data.threshold < 0) {
+            showToast(translate("Threshold should be greater then 0."));
```
Comment (ymaheshwari1) [inline src/components/RuleItem.vue:164]:
```suggestion
            showToast(translate("Threshold should be greater then or equal to 0."));
```

File: src/components/RuleItem.vue [hunk 4]
```diff
@@ -206,8 +211,12 @@ async function editSafetyStock() {
     },
     {
       text: translate('Update'),
-      handler: async (data) => {
+      handler: async (data: any) => {
         if(data.safetyStock) {
+          if(data.safetyStock < 0) {
```
Comment (ymaheshwari1) [inline src/components/RuleItem.vue:216]: This check for safetyStock value is redundant. We can move this check above the parent `if`.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Pull request merged**